formbricks/formbricks
# Add Data Service for Responses, Webhooks, Surveys & Api Keys

## What does this PR do?



• Add data service and zod types for webhooks, surveys & api Keys
• Add error classes for common service errors
• Rewrite responses service by using new error classes and thus improving performance, readability and response codes
• Add helper function for pipeline calls
• Rewrite webhook endpoints using services and error codes



## Type of change




## How should this be tested?



As the responses (client) and webhook endpoints are affected please check them:
- create response (link survey + client)
- update response (link survey + client)
- create, get, delete and list webhooks (api)
- webhooks are triggered (responseCreated, responseUpdated, responseFinished)

## Checklist
